packages/unified-latex-to-pretext/libs/pre-conversion-subs/environment-subs.ts

```diff
@@ -129,4 +172,85 @@ export const environmentReplacements: Record<
+        { statement: boolean; aliases: string[] }
+    > = {
+        abstract: { statement: false, aliases: ["abs", "abstr"] },
+        acknowledgement: { statement: false, aliases: ["ack"] },
+        algorithm: { statement: true, aliases: ["algo", "alg"] },
+        assumption: { statement: true, aliases: ["assu", "ass"] },
+        axiom: { statement: true, aliases: ["axm"] },
+        claim: { statement: true, aliases: ["cla"] },
+        conjecture: { statement: true, aliases: ["con", "conj", "conjec"] },
+        construction: { statement: false, aliases: [] },
+        convention: { statement: false, aliases: ["conv"] },
+        corollary: {
+            statement: true,
+            aliases: ["cor", "corr", "coro", "corol", "corss"],
+        },
+        definition: {
+            statement: true,
+            aliases: ["def", "defn", "dfn", "defi", "defin", "de"],
+        },
+        example: {
+            statement: true,
+            aliases: ["exam", "exa", "eg", "exmp", "expl", "exm"],
+        },
+        exercise: { statement: true, aliases: ["exer", "exers"] },
+        exploration: { statement: false, aliases: [] },
+        fact: { statement: true, aliases: [] },
+        heuristic: { statement: true, aliases: [] },
+        hypothesis: { statement: true, aliases: ["hyp"] },
+        identity: { statement: true, aliases: ["idnty"] },
+        insight: { statement: false, aliases: [] },
+        investigation: { statement: false, aliases: [] },
+        lemma: { statement: true, aliases: ["lem", "lma", "lemm", "lm"] },
+        notation: {
+            statement: false,
+            aliases: ["no", "nota", "ntn", "nt", "notn", "notat"],
+        },
+        note: { statement: false, aliases: ["notes"] },
+        observation: { statement: false, aliases: ["obs"] },
+        principle: { statement: true, aliases: [] },
+        problem: { statement: true, aliases: ["prob", "prb"] },
+        project: { statement: false, aliases: [] },
+        proof: { statement: false, aliases: ["pf", "prf", "demo"] },
+        proposition: {
+            statement: true,
+            aliases: ["prop", "pro", "prp", "props"],
+        },
+        question: { statement: true, aliases: ["qu", "ques", "quest", "qsn"] },
+        remark: {
+            statement: false,
+            aliases: ["rem", "rmk", "rema", "bem", "subrem"],
+        },
+        task: { statement: true, aliases: [] },
+        theorem: {
+            statement: true,
+            aliases: ["thm", "theo", "theor", "thmss", "thrm"],
+        },
+        warning: { statement: false, aliases: ["warn", "wrn"] },
+    };
+    // For each environment PreTeXt has, we create entries for `environmentReplacements` using all reasonable aliases
+    for (const [env, obj] of Object.entries(envAliases)) {
+        reps[env] = envFactory(env, obj.statement);
+        for (const alias of obj.aliases) {
+            reps[alias] = envFactory(env, obj.statement);
+        }
+    }
```

Comment:
I feel like this could be a little nicer in a functional style. Something like

```
const expandedEnvAliases = envAliases
  .entries()
  .flatMap(([env, spec]) => [
    [env, {statement: env.statement}], 
    ...env.aliases.map(name => [name, {statement: env.statement}])
   ]);
return Object.fromEntries(
  expandedEnvAliases.map(
    ([env, {statement}]) => [env, envFactory(env, statement)]
  )
);
```

Reply:
Good idea.  I forget about this magic.